pkg/sloop/processing/eventcount.go

```diff
@@ -49,13 +48,27 @@ func updateEventCountTable(
 		return nil
 	}
 
-	// Truncate long-lived events to (watch.Timestamp - maxLookback).  This avoids filling in data that will immediately
-	// be garbage collected, and avoids creating transactions that are too large and fail
-	watchTs, err := ptypes.Timestamp(watchRec.Timestamp)
-	if err != nil {
-		return err
+	// Truncate long-lived events to available partitions
+	// This avoids filling in data that will go beyond the current time range
+
+	// Default truncate as computedLastTs -1 * PartitionDuration.
+	// Essentially only allowing events in 1 partition by default.
+	truncateTs := computedLastTs.Add(-1 * untyped.GetPartitionDuration())
```

Comment:
Minor: this should only happen for the first event.  Its a little odd that it computes lastTs-1h where everything else is on a partition boundary.  If we want to be more consistent, we could compute the partition start for lastTs and use that - then we consistently truncate on partition boundaries.  But we could also leave it as-is, since its just one event processing.

For the comment we may want to make it more clear that this will only happen for 1 event on a new sloop with no data.